zetaclient/chains/solana/signer/increment_nonce.go

```diff
@@ -0,0 +1,108 @@
+package signer
+
+import (
+	"context"
+
+	"cosmossdk.io/errors"
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/near/borsh-go"
+
+	contracts "github.com/zeta-chain/node/pkg/contracts/solana"
+	"github.com/zeta-chain/node/x/crosschain/types"
+)
+
+// createAndSignMsgIncrementNonce creates and signs a increment_nonce message for gateway increment_nonce instruction with TSS.
+func (signer *Signer) createAndSignMsgIncrementNonce(
```

Comment:
Consider calling this `createAndSignTssMsgIncrementNonce`

NIT: it might be good to create a msg and then call two functions `SignTSS` and `SignRelayer` in `func prepareIncrementNonceTx` . It would make unit testing easier.